### PR TITLE
[js_hw4_arrays][Task_01-Task_03] Accomlished all tasks

### PR DESCRIPTION
**Pull request by Maximus2806**

### CODE REVIEW

File: js_hw_4/task_01.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+// 2. В функции создайте массив имен сотрудников 5шт (Имена придумать самим)
+// 3. В функции создайте массив зарплат сотрудников 5 шт(цифры придумать самим)
+// 4. Функция должна принимать 1 аргумент - имя сотрудника
+// 5. Функция должна возвращать новый массив, где первый элемент - имя сотрудника, второй - его зарплата
+// 6. Для поиска ответа функции нужно найти индекс сотрудника в массиве имен. Зарплату взять с ТЕМ ЖЕ индексом что и имя
+// 7. Для возврата из функции создайте массив, методом .push поместите в него имя и зарплату, и через return верните созданный массив
+// 8. Если такое имя сотрудника в массиве не найдется - вернуть null
+
+function getEmployeeInfo(name) {
+    if(name!== undefined && name!== null && name.length && !+name){
```
Comment (VladimirLQA) [inline js_hw_4/task_01.js]:
```suggestion
    if(!name || !name.trim().length) { /* trow error */}
```

File: js_hw_4/task_01.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        const names = ["Bill", "Alex", "Anna", "Jhonn", "Antony"];
+    const salaries = [1500, 3000, 1200, 1700, 850];
+    const index = names.indexOf(name);
```
Comment (VladimirLQA) [inline js_hw_4/task_01.js:14]:
```suggestion
    const index = names.indexOf(name);
    if(index === -1) return null
```

File: js_hw_4/task_01.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    let resultArr = [];
+    for (item of names) {
```
Comment (VladimirLQA) [inline js_hw_4/task_01.js:16]: Use let or const

File: js_hw_4/task_02.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
+// 1. У вас есть массив названий пицц вашего конкурента. Создайте функцию, которая будет принимать ваш набор названий пицц (массив) 
+//   и возвращать только те, которых нет у конкурента (тоже массив). Если все ваши пиццы есть у конкурента - вернуть null
+//   Пиццы конкурента:
+//   const competitorPizzas = ['Peperoni', 'Caprichosa', 'Diablo', '4 cheeses', 'hawai']
+  
+
+
+const competitorPizzas = ['Peperoni', 'Caprichosa', 'Diablo', '4 cheeses', 'hawai'];
+function checkCompetitorsPizzas(myPizzaArr) {
```
Comment (VladimirLQA) [inline js_hw_4/task_02.js:9]: Function must be case insensitive

File: js_hw_4/task_02.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+console.log(checkCompetitorsPizzas(pizzaSet));
+
+// 2. Написать функцию, которая принимает предложение (слова разделенные только пробелами) в качестве параметра и выводит в консоль слово с наибольшим количеством букв. 
+//   Если таких слов несколько - выводит их все.
+
+function findLongest(sentence) {
+    const wordsArr = sentence.split(' ');
+    let longestWord = "";
+    for (let i = 0; i < wordsArr.length - 1; i++) {
```
Comment (VladimirLQA) [inline js_hw_4/task_02.js]: Possible to refactor this solution to use only one loop

**Review by VladimirLQA — CHANGES_REQUESTED** [review]

File: js_hw_4/task_01.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        if (item === name){
+            resultArr.push(item);
+            resultArr.push(salaries[index]);
+        }
+    }
+    return index === -1 ? null: resultArr
```
Comment (VladimirLQA) [inline js_hw_4/task_01.js]:
```suggestion
    return [names[index], salaries[index]]
```

File: js_hw_4/task_03.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+        }    
+    }
+}
+    
+console.log(sumOfDigits("19"));
+
+// 6*. Написать функцию, которая принимает на вход строку с текстом, и заменяет каждую пару стоящих подряд идентичных букв на одну следующую в алфавите, 
+//     и так пока в тексте не останется двух одинаковых букв стоящих рядом (через пробел и другой знак препинания можно)
+//     Пример: aabc => bbc => cc => d
+function replaceAllDoublications(str) {
```
Comment (VladimirLQA) [inline js_hw_4/task_03.js:28]: Try with this cases
```js
console.log(replaceAllDuplications('edaabc')); // f
console.log(replaceAllDuplications('daabc'));  // e
console.log(replaceAllDuplications('baac'));   // d
```

File: js_hw_4/task_01.js (hunk 1 above)
Comment (Maximus2806) [reply js_hw_4/task_01.js]: @VladimirLQA your solution will throw "TypeError: name.trim is not a function" when the number is provided in input

File: js_hw_4/task_01.js (hunk 2 above)
Comment (Maximus2806) [reply js_hw_4/task_01.js:14]: What's wrong with my implementation? 
return index === -1 ? null : resultArr

File: js_hw_4/task_02.js (hunk 4 above)
Comment (Maximus2806) [reply js_hw_4/task_02.js:9]: Implemented by adding competitorPizzas.map(word => word.toLowerCase())

File: js_hw_4/task_02.js (hunk 5 above)
Comment (Maximus2806) [reply js_hw_4/task_02.js]: Done

File: js_hw_4/task_01.js (hunk 2 above)
Comment (VladimirLQA) [reply js_hw_4/task_01.js:14]: If index is `-1` immediately return the `null`, no need to do another steps
You push undefined to result array `salaries[-1]`

File: js_hw_4/task_01.js (hunk 1 above)
Comment (VladimirLQA) [reply js_hw_4/task_01.js]: So you can add check on typeof for strings + empty string with logical `and` operator

File: js_hw_4/task_01.js (hunk 2 above)
Comment (Maximus2806) [reply js_hw_4/task_01.js:14]: I got it. Thank you for explanation!

File: js_hw_4/task_03.js (hunk 7 above)
Comment (Maximus2806) [reply js_hw_4/task_03.js:28]: @VladimirLQA Resolved. Check please

File: js_hw_4/task_03.js [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+//   После окончания сложений возвращает полученное число. Например при подаче числа 19 (1+9=10>9, потому 1+0=1) выводится 1
+
+function sumOfDigits(num) {
+    if (!+num){
+        return "The argument must be a number!";
+    } else {
+        const numStr = Math.abs(num).toString();
+        const ArrOfDigits = numStr.split('');
+        let sum = 0;
+        for (digit of ArrOfDigits) {
```
Comment (anatoly-karpovich) [inline js_hw_4/task_03.js:12]:
```suggestion
        const arrOfDigits = numStr.split('');
        let sum = 0;
        for (digit of arrOfDigits) {
```
only Classes should be named started with the capital letter

**Review by anatoly-karpovich — APPROVED** [review]

File: js_hw_4/task_01.js (hunk 2 above)
Comment (Maximus2806) [reply js_hw_4/task_01.js:14]: @VladimirLQA please check the final solution cause it blocks merging
function getEmployeeInfo(name) {    
    if (!name || !name.length){ throw new Error("Invalid input")}
    const names = ["Bill", "Alex", "Anna", "Jhonn", "Antony"];
    const salaries = [1500, 3000, 1200, 1700, 850];
    const index = names.indexOf(name.trim());
    if (index === -1) return null;
    return [names[index], salaries[index]];
}

**Review by VladimirLQA — APPROVED** [review]

File: js_hw_4/task_03.js (hunk 8 above)
Comment (Maximus2806) [reply js_hw_4/task_03.js:12]: Will be fixed and delivered in the next commit with next homework

**Pull request merged**